Add librosco (0.1.11)

colinbourassa opened the pull request.

starius commented on src/librosco.mk:
```diff
@@ -0,0 +1,20 @@
+define $(PKG)_BUILD
+    mkdir '$(1)/build'
+    cd '$(1)/build' && cmake .. \
+        -DCMAKE_TOOLCHAIN_FILE='$(CMAKE_TOOLCHAIN_FILE)' \
```
Replace `cmake .. -DCMAKE_TOOLCHAIN_FILE='$(CMAKE_TOOLCHAIN_FILE)'` with `'$(TARGET)-cmake'`

starius commented on src/librosco.mk:
```diff
@@ -0,0 +1,20 @@
+        -DCMAKE_TOOLCHAIN_FILE='$(CMAKE_TOOLCHAIN_FILE)' \
+        -DBUILD_STATIC=$(if $(BUILD_STATIC),ON,OFF)
+
+    $(MAKE) -C '$(1)/build' -j '$(JOBS)' install
```
Please separate building and installing to two commands. Build with `-j '$(JOBS)'`, install with `-j 1`.

starius commented:
Please provide an updater for this package.

```makefile
define $(PKG)_UPDATE
    $(call MXE_GET_GITHUB_TAGS, colinbourassa/librosco, v)
endef
```

starius commented:
Files installed:

```
i686-w64-mingw32.static/bin/librosco.a
i686-w64-mingw32.static/bin/readmems.exe
i686-w64-mingw32.static/doc/librosco/LICENSE.TXT
i686-w64-mingw32.static/doc/librosco/README.TXT
i686-w64-mingw32.static/include/rosco.h
i686-w64-mingw32.static/include/rosco_version.h
i686-w64-mingw32.static/installed/librosco
```

Problems:

  * you should not install `*.a` files to `bin/`. Install `*.a` (and `*.dll.a`) to `lib/` (static target). Install `*.dll` to `bin` (shared target). I haven't built other targets yet. Please make sure they install the library in proper format to proper directory.
  * Is readmems.exe needed for using this library? Probably no. Rename it to `test-librosco-readmems.exe` when installing.
  * Please provide `*.pc` file to facilitate the use of the library. You can use files from `usr/i686-w64-mingw32.static/lib/pkgconfig/` as examples. See also https://secure.freedesktop.org/~dbn/pkg-config-guide.html
  * do not install documentation.

colinbourassa commented:
Thank you for the comments. I believe I've addressed all of them in the latest commit (which I've squashed into the previous) and in a newer version of the upstream package.

starius commented:
> which I've squashed into the previous

There are 3 commits: initial one, changes and merge. Can you them with a single commit, please?

colinbourassa commented:
Sorry about that; I didn't do a forced push. Should be fixed now.

starius commented:
I think, commit message should be replaced with "Add librosco (0.1.8)".

colinbourassa commented:
You're right. I've corrected the commit message as well.

starius commented:
I have rebuilt it. Now it installs the following files:

```
usr/i686-w64-mingw32.static/include/rosco.h
usr/i686-w64-mingw32.static/include/rosco_version.h
usr/i686-w64-mingw32.static/installed/librosco
usr/i686-w64-mingw32.static/lib/librosco.a
usr/i686-w64-mingw32.static/lib/pkgconfig/librosco.pc
```

Could you provide a sample application using this library (using pkgconfig is preferred) and install it to `'$(PREFIX)/$(TARGET)/bin/test-librosco.exe'`, please?

colinbourassa commented:
I've amended the commit to use librosco 0.1.9, which now installs its test application as 'test-librosco.exe' to the bin directory.

starius commented:
It built successfully on all 4 targets and created the following files:

  * i686-w64-mingw32.shared/bin/librosco.dll
  * i686-w64-mingw32.shared/bin/test-librosco.exe
  * i686-w64-mingw32.shared/include/rosco.h
  * i686-w64-mingw32.shared/include/rosco_version.h
  * i686-w64-mingw32.shared/installed/librosco
  * i686-w64-mingw32.shared/lib/pkgconfig/librosco.pc
  * i686-w64-mingw32.static/bin/test-librosco.exe
  * i686-w64-mingw32.static/include/rosco.h
  * i686-w64-mingw32.static/include/rosco_version.h
  * i686-w64-mingw32.static/installed/librosco
  * i686-w64-mingw32.static/lib/librosco.a
  * i686-w64-mingw32.static/lib/pkgconfig/librosco.pc
  * x86_64-w64-mingw32.shared/bin/librosco.dll
  * x86_64-w64-mingw32.shared/bin/test-librosco.exe
  * x86_64-w64-mingw32.shared/include/rosco.h
  * x86_64-w64-mingw32.shared/include/rosco_version.h
  * x86_64-w64-mingw32.shared/installed/librosco
  * x86_64-w64-mingw32.shared/lib/pkgconfig/librosco.pc
  * x86_64-w64-mingw32.static/bin/test-librosco.exe
  * x86_64-w64-mingw32.static/include/rosco.h
  * x86_64-w64-mingw32.static/include/rosco_version.h
  * x86_64-w64-mingw32.static/installed/librosco
  * x86_64-w64-mingw32.static/lib/librosco.a
  * x86_64-w64-mingw32.static/lib/pkgconfig/librosco.pc

The problem is that it is not usable on shared targets because
file `librosco.dll.a` is not installed to directory `lib/`:

```
$ ./usr/bin/i686-w64-mingw32.shared-gcc /tmp/readmems.c -o /tmp/readmems.exe `./usr/bin/i686-w64-mingw32.shared-pkg-config --libs librosco`
/home/mxe/mxe-librosco/usr/bin/i686-w64-mingw32.shared-ld: cannot find -lrosco
collect2: error: ld returned 1 exit status
```

Please do the following:

  * build the test with the command above. Adjust it for MXE:
```
'$(TARGET)-gcc' $(1)/src/readmems.c \
    -o '$(PREFIX)/$(TARGET)/bin/test-librosco.exe' \
    `'$(TARGET)-pkg-config' --libs librosco`
```
  * make sure `make librosco` fails with the error "cannot find -lrosco"
  * adjust build rules of librosco to install `librosco.dll.a` to
    directory `lib/` on shared targets
  * rebase and squash this branch.

colinbourassa commented:
Ah, I see. I've updated the package to install librosco.dll.a, and I updated the recipe to build test-librosco.exe by using pkg-config, as you demonstrated.

starius commented:
Looks good to me! Please rebase it with `git pull --rebase https://github.com/mxe/mxe master`

colinbourassa commented:
Done.

starius commented:
Thanks!
